sqlglot/dialects/tsql.py

```diff
@@ -370,6 +370,17 @@ def _timestrtotime_sql(self: TSQL.Generator, expression: exp.TimeStrToTime):
     return sql
 
 
+def _parse_datetrunc(args: t.List) -> exp.TimestampTrunc:
+    unit = seq_get(args, 0)
+    this = seq_get(args, 1)
+
+    # Every date - time type in tsql resoves to DATETIME2
+    if isinstance(this, exp.Expression) and this.is_string:
+        this = exp.TimeStrToTime(this=this)
```

Comment:
Not sure if this should be `exp.TimeStrToTime` here, I think we could go with `exp.cast(this, exp.DataType.Type.DATETIME2)`

Reply:
Ok, I thought `exp.TimeStrToTime` was a better approach, but I will change it to the explicit cast.

Reply:
You'd have to see if `exp.TimeStrToTime` behaves as expected in T-SQL, i.e does it cast to `DATETIME2`, does it have other side-effects etc. 

Otherwise, always try to go with the straightforward solution